src/main/java/org/cbioportal/web/columnar/ClinicalDataBinner.java

```diff
@@ -26,6 +26,26 @@ public ClinicalDataBinner(
         this.dataBinner = dataBinner;
     }
 
+    // TODO move this to a utility class?
+    public List<ClinicalData> convertCountsToData(List<ClinicalDataCount> clinicalDataCounts)
+    {
+        return clinicalDataCounts
+            .stream()
+            .map(c -> {
+                // TODO get rid of the for loop and use something nicer and more efficient?
+                List<ClinicalData> data = new ArrayList<>(c.getCount());
+                for (int i=0; i < c.getCount(); i++) {
+                    ClinicalData d = new ClinicalData();
+                    d.setAttrId(c.getAttributeId());
+                    d.setAttrValue(c.getValue());
+                    data.add(d);
+                }
+                return data;
+            })
+            .flatMap(Collection::stream)
+            .toList();
+    }
+    
```

Comment:
I think atleast for now the cleaner way would to have another function.

private List<ClinicalData> generateClinicalDataFromClinicalDataCounts(ClinicalDataCounts counts) { 
  nasty for loop
}

So then here you can just do this
`clinicalDataCounts
            .stream()
            .map(this::generateClinicalDataFromClinicalDataCounts)
            .flatMap(Collection::stream)
            .toList();`

Reply:
```suggestion
private List<ClinicalData> generateClinicalDataFromClinicalDataCounts(ClinicalDataCounts counts) {
nasty for loop
}

    clinicalDataCounts .stream() .map(this::generateClinicalDataFromClinicalDataCounts) .flatMap(Collection::stream) .toList();
```